ysh/func_proc.py

```diff
@@ -369,15 +351,31 @@ def _BindTyped(
         if rest:
             lval = LeftName(rest.name, rest.blame_tok)
 
-            rest_val = value.List(pos_args[num_params:])
-            mem.SetLocalName(lval, rest_val)
+            rest_val = pos_args[num_params:-1] if block_param else pos_args[num_params:]
```

Comment:
Hmm why move `if block_param` after the `rest` section?  

Wouldn't keeping it before that simplify this condition?